go/mysql/conn.go

```diff
@@ -1707,7 +1707,15 @@ func (c *Conn) IsMarkedForClose() bool {
 	return c.closing
 }
 
+func (c *Conn) IsShuttingDown() bool {
+	return c.listener.shutdown.Load()
+}
+
 // GetTestConn returns a conn for testing purpose only.
 func GetTestConn() *Conn {
 	return newConn(testConn{})
 }
+
+func GetTestServerConn(listener *Listener) *Conn {
```

Comment:
I know the function already says `Test` but maybe it is a good idea to add a comment something like - 
```
// GetTestServerConn is only meant to be used for testing. It creates a server connection using a testConn and the provided listener.
```
And I would move this function to the `conn_fake.go` file. That files defines the `testConn` struct too and it contains all the test code we keep in the package.

Reply:
@GuptaManan100 Thanks! I moved this and the other `GetTestConn()` function to `conn_fake.go`.